lib/system_checks.py

```diff
@@ -0,0 +1,112 @@
+# This file handles the checking of the system
+# There is a list of checks that is made up of tuples structured the following way:
+# - the function to call to check. This will return True or None for success and False for failure
+# - What severity the False return value has. If the Status is Error we raise and exit the GMT
+# - A string what is being checked
+# - A string to output on WARN or INFO
+# It is possible for one of the checkers or metric providers to raise an exception if something should fail specifically
+# otherwise you can just return False and set the Status to ERROR for the program to abort.
+
+import sys
+import os
+from enum import Enum
+import psutil
+
+from global_config import GlobalConfig
+from lib.terminal_colors import TerminalColors
+import utils
+
+Status = Enum('Status', ['ERROR', 'INFO', 'WARN'])
+
+class ConfigurationCheckError(Exception):
+    pass
+
+######## CHECK FUNCTIONS ########
+def check_metric_providers(runner):
+    for mp in runner._Runner__metric_providers:
+        if hasattr(mp, 'check_system'):
+            if mp.check_system() == False:
+                return False
+
+def check_one_psu_provider(_):
+    metric_providers = list(utils.get_metric_providers(GlobalConfig().config).keys())
+    if sum(True for provider in metric_providers if ".energy" in provider and ".machine" in provider) > 1:
+        return False
+
+def check_tmpfs_mount(_):
+    for mp in psutil.disk_partitions():
+        if mp.mountpoint == '/tmp' and mp.fstype != 'tmpfs':
+            return False
+
+def check_free_disk(percent):
+    # We are assuming that the GMT is installed on the system partition!
+    usage = psutil.disk_usage(os.path.abspath(__file__))
+    if usage.percent >= percent:
+        return False
+
+def check_free_disk_80(_):
+    return check_free_disk(80)
+
+def check_free_disk_90(_):
+    return check_free_disk(90)
+
+def check_free_disk_95(_):
+    return check_free_disk(95)
+
+def check_temperature(_):
```

Comment:
This is checking for the critical temp? Does this make sense? To my understanding the PC turns off when that is reached ...?

Reply:
It checks for high temp not critical. Critical is another field.